### ISSUE TITLE
Add overlay+package to the Nix Flake

### ISSUE THREAD

**Issue by toastal** [purescm/purescm#220]
Despite Nix being used, the Nix Flake doesn’t actually expose the package instead suggesting going thru Microsoft’s npm, the JavaScript package manager, for distribution. I would like to change that so npm can be skipped, which makes sense for a backend not targeting JavaScript.

**Comment by f-f** [comment]
We are not focused on distribution yet, but it the packaging is not too bulky we can have a look at it

**Comment by f-f** [comment]
We can try it out - the main thing we'd need is that there's a way to test this in CI and make sure it doesn't break over time

**Comment by f-f** [comment]
[purescript-overlay](https://github.com/thomashoneyman/purescript-overlay) is just using NPM packages and I think that's an ok solution, so I guess that's no.2?

**Comment by f-f** [comment]
The new Spago could also do with being added to Nixpkgs - there's [a PR in flight](https://github.com/purescript/spago/pull/981) about building it locally with a flake, but I think it's going to be less hassle to just move that to nixpkgs where there's folks that know more Nix than me

**Comment by f-f** [comment]
> Does `Exception: library (Effect.Console foreign) not found` mean anything to you?

Yeah it looks like it's not finding the folder with the compiled scheme files. How's your setup looking? Do you have a thing somewhere I can run and look at?

**Comment by f-f** [comment]
Ah I see now. The readme is missing an important step - you need to point the project [to the chez package set](https://raw.githubusercontent.com/purescm/purescm/master/package-sets/1.0.0.json), since the default package set will only have JS FFI. It now makes sense that the thing is complaining about not finding the foreign file 🙂 

```yaml
workspace:
  package_set:
    url: https://raw.githubusercontent.com/purescm/purescm/master/package-sets/1.0.0.json
```
